### PR TITLE
Add checksum for py-markdown 3.4.1

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-markdown/package.py [hunk 1]
```diff
@@ -36,7 +37,9 @@ class PyMarkdown(PythonPackage):
...
     depends_on("python@2.7:2.8,3.3.5:", when="@3.1.1:")
     depends_on("python@3.6:", when="@3.3.4:")
+    depends_on("python@3.7:", when="@3.4.1:")
 
-    depends_on("py-setuptools", type="build", when="@2.6.11:")
-    depends_on("py-setuptools@36.6:", type="build", when="@3.1:")
-    depends_on("py-importlib-metadata", type=("build", "run"), when="@3.3.4: ^python@:3.7")
+    depends_on("py-setuptools", when="@2.6.11:", type="build")
+    depends_on("py-setuptools@36.6:", when="@3.1:", type="build")
+    depends_on("py-importlib-metadata", when="@3.3.4: ^python@:3.7", type=("build", "run"))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-markdown/package.py:44]: Thanks for reordering the args.  Reads better.

File: var/spack/repos/builtin/packages/py-markdown/package.py [hunk 2]
```diff
@@ -17,6 +17,7 @@ class PyMarkdown(PythonPackage):
     homepage = "https://python-markdown.github.io/"
     pypi = "markdown/Markdown-2.6.11.tar.gz"
 
+    version("3.4.1", sha256="3b809086bb6efad416156e00a0da66fe47618a5d6918dd688f53f40c8e4cfeff")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-markdown/package.py:20]: Confirmed the sha256.

File: var/spack/repos/builtin/packages/py-markdown/package.py [hunk 3]
```diff
@@ -36,7 +37,9 @@ class PyMarkdown(PythonPackage):
     depends_on("python@2.7:2.8,3.2:3.6", when="@2.6.8:2.6.11")
     depends_on("python@2.7:2.8,3.3.5:", when="@3.1.1:")
     depends_on("python@3.6:", when="@3.3.4:")
+    depends_on("python@3.7:", when="@3.4.1:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-markdown/package.py]: These should probably all be `type=("build", "run")`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-markdown/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-markdown/package.py:44]: @tldahlgren I can do a batch update of all packages (it's a simple `sed`) if everyone agrees and if it goes into a single PR...

**Pull request merged**